src/averageAges.js

```diff
@@ -59,6 +113,20 @@ function calculateAverageAgeDiff(people, onlyWithSon) {
   // 2. keep people who have mothers in the array
   // 3. calculate the difference child.born - mother.born
   // 4. return the average value
+
+  const childs = people.filter(child =>
+    onlyWithSon
+      ? people.find(human => human.name === child.mother) && child.sex === 'm'
```

Comment:
* Consider creating helper functions and constants variables before your task functions and reuse them inside calculateWomenAverageAge, calculateMenAverageAge, etc.
* Consider using constants instead of magic numbers, it's an antipattern